src/hooks/useNewsApi.ts

```diff
@@ -36,37 +37,38 @@ const useNewsApi = () => {
     [navigate],
   );
 
-  const fetchNewsApi = useCallback(async () => {
-    setLoading(true);
+  const fetchNewsApi = useCallback(
+    async (signal: AbortSignal) => {
+      setLoading(true);
 
-    const shouldCreateNewInstance = api.current.apiSlug !== source;
-    const apiToUse = shouldCreateNewInstance ? ApiFactory.createApi(source) : api.current;
+      const shouldCreateNewInstance = api.current.apiSlug !== source;
+      const apiToUse = shouldCreateNewInstance ? ApiFactory.createApi(source) : api.current;
```

Comment:
I think logic around creating new api instance does not belong here.

Reply:
Currently I have no ideas how to implement the change of the api differently =(